lib/validators/ProfileCreation.js

```diff
@@ -50,8 +54,34 @@ export default Joi.object({
         .empty('')
         .valid(...allowedMinors)
     ),
-  graduationDate: Joi.string().required().max(maxGradDateLength), // TODO: require a season and a year, make sure year is in the future
-  gpa: Joi.string().required(), // TODO: limit range of float/int
+  graduationDate: Joi.string()
+    .max(maxGradDateLength)
+    .pattern(/^((Spring)|(Summer)|(Fall)|(Winter)\b) ([0-9]{4})$/) //(i?)
+    .custom((value) => {
+      console.log('current year: ' + currentYear + ' and input year: ' + value);
+
+      if (
+        !value.includes(currentYear) &&
+        !value.includes(currentYear + 1) &&
+        !value.includes(currentYear + 2) &&
+        !value.includes(currentYear + 3) &&
+        !value.includes(currentYear + 4) &&
+        !value.includes(currentYear + 5) &&
+        !value.includes(currentYear + 6) &&
+        !value.includes(currentYear + 7) &&
+        !value.includes(maxGradYear)
+      ) {
+        throw new Error('Graduation Year must be between ' + currentYear + ' and ' + maxGradYear);
+      } else {
+        return value;
+      }
+    })
+    .required(), // TODO: require a season and a year, make sure year is in the future
+  gpa: Joi.string()
+    .max(5)
+    .min(2)
+    .pattern(/^([0-3]\.\d\d?\d?\d?)|(4\.00?0?0?)$/)
```

Comment:
This regex is wrong. According to https://regex101.com, it's like `(/^([0-3]\.\d\d?\d?\d?))|((4\.00?0?0?)$)/`, so a match either starts with `([0-3]\.\d\d?\d?\d?)` or ends with `(4\.00?0?0?)` (This precedence of the regex operators is kind of unintuitive, and I thought your regex would work too).

Once you fix this, make majorGpa be validated with the same function.